projects/Mallard/src/components/front/cards.tsx

```diff
@@ -18,24 +20,28 @@ const styles = StyleSheet.create({
         flexBasis: '100%',
     },
 })
+interface SmallCardProps {
+    style: StyleProp<ViewStyle>
+    article: FrontArticle
+}
+type InjectedProps = NavigationInjectedProps<Params>
 const SmallCard = withNavigation(
     ({
         style,
-        headline,
-        kicker,
-        path,
+        article,
         navigation,
-    }: NavigationInjectedProps & {
-        style: StyleProp<ViewStyle>
-        headline: string
-        kicker: string
-        path: string
-    }) => {
+    }: SmallCardProps & Partial<InjectedProps>) => {
         const { appearance } = useArticleAppearance()
+        if (navigation == null) {
+            throw new Error('No navigation presetn in cards.')
+        }
         return (
             <View style={style}>
                 <Highlight
-                    onPress={() => navigation.navigate('Article', { path })}
+                    onPress={() => {
+                        console.log(article)
+                        navigation.navigate('Article', { article: article })
```

Comment:
I wonder if we want this route to demand a full article rather than take something like `{path, partialArticleDataFromFront?}`. This would allow the route to function on its own just from an article URL (we will appreciate that when making the web version) and taking partial article data lets us show something while the full article loads instead than a loading screen

Reply:
ah, I've had a closer look. it seems we already do this with `FrontArticle`. Nice! I'd still separate the path from `FrontArticle` to not have to pass all of that to the view since it's just aesthetic

Reply:
Currently we do, as the front needs to supply some of these things. We'll need to think how deep linking should work in this context.

Reply:
Sorry, you mean the full article request doesn't contain everything we need to render the article? we need `FrontArticle` + `Article`? I don't think that's very clean :/ `Article` should be self container

Reply:
This would be significant to implement in the backend and will need to be after we've discussed the api with tools.

Reply:
Shall we add a Trello card to remind us to pick this up after the API discussion?